mobile/src/main/java/org/openhab/habdroid/ui/LogActivity.java

```diff
@@ -0,0 +1,137 @@
+package org.openhab.habdroid.ui;
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ScrollView;
+import android.widget.TextView;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import org.openhab.habdroid.R;
+import org.openhab.habdroid.util.Constants;
+import org.openhab.habdroid.util.Util;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+
+import static org.openhab.habdroid.util.Util.getHostFromUrl;
+
+public class LogActivity extends AppCompatActivity {
+    private static final String TAG = LogActivity.class.getSimpleName();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        Util.setActivityTheme(this);
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_log);
+
+        Toolbar toolbar = findViewById(R.id.openhab_toolbar);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        FloatingActionButton fab = findViewById(R.id.shareFab);
+        TextView logView = findViewById(R.id.log);
+        fab.setOnClickListener(v -> {
+            Intent sendIntent = new Intent();
+            sendIntent.setAction(Intent.ACTION_SEND);
+            sendIntent.putExtra(Intent.EXTRA_TEXT, logView.getText());
+            sendIntent.setType("text/plain");
+            startActivity(sendIntent);
+        });
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        setProgressIndicatorVisible(true);
+        new GetLogFromAdbTask().execute(false);
+    }
+
+    private void setProgressIndicatorVisible(boolean visible) {
+        findViewById(R.id.progressBar).setVisibility(visible ? View.VISIBLE : View.GONE);
```

Comment:
Save a reference to the progress bar in onCreate? Should probably also toggle visibility of the log view here, and either toggle visibility or enabled state of the FAB.

Reply:
Done.